barretenberg/cpp/src/barretenberg/honk/proof_system/protogalaxy_prover.cpp

```diff
@@ -1,54 +1,52 @@
 #include "protogalaxy_prover.hpp"
 #include "barretenberg/proof_system/flavor/flavor.hpp"
 namespace proof_system::honk {
-template <class Flavor>
-ProtoGalaxyProver_<Flavor>::ProtoGalaxyProver_(std::vector<std::shared_ptr<Instance>> insts)
-    : instances(insts)
-{}
 
 /**
  * @brief Prior to folding we need to add all the public inputs to the transcript, labelled by their corresponding
  * instance index, compute all the instance's polynomials and record the relation parameters involved in computing these
  * polynomials in the transcript.
  *
  */
-template <class Flavor> void ProtoGalaxyProver_<Flavor>::prepare_for_folding()
+template <class ProverInstances> void ProtoGalaxyProver_<ProverInstances>::prepare_for_folding()
 {
-    for (const auto& instance : instances) {
+    // this doesnt work in the current format
+    auto idx = 0;
+    for (auto it = instances.begin(); it != instances.end(); it++, idx++) {
+        auto instance = *it;
         instance->initialise_prover_polynomials();
 
-        const auto instance_index = std::to_string(instance->index);
+        auto domain_separator = std::to_string(idx);
```

Comment:
we still need to domain separate information coming from different instances but now the index is not part of ProverInstance and VerifierInstance. This could be further cleaned up when our proofs become data structures (#656)